In [1]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 100)

In [2]:
train = pd.read_csv("../data/raw/train.csv")
test = pd.read_csv("../data/raw/test.csv")
store = pd.read_csv("../data/raw/store.csv")

/var/folders/vk/6pn68sk92z93wvsph2msbyvh0000gn/T/ipykernel_8336/4109812108.py:1: DtypeWarning: Columns (0: StateHoliday) have mixed types. Specify dtype option on import or set low_memory=False.
  train = pd.read_csv("../data/raw/train.csv")


In [3]:
print("Train shape:", train.shape)
print("Test shape:", test.shape)
print("Store shape:", store.shape)

Train shape: (1017209, 9)
Test shape: (41088, 8)
Store shape: (1115, 10)


In [4]:
train.head()

,Store,DayOfWeek,Date,Sales,Customers,Open,Promo,StateHoliday,SchoolHoliday
0,1,5,2015-07-31,5263,555,1,1,0,1
1,2,5,2015-07-31,6064,625,1,1,0,1
2,3,5,2015-07-31,8314,821,1,1,0,1
3,4,5,2015-07-31,13995,1498,1,1,0,1
4,5,5,2015-07-31,4822,559,1,1,0,1


In [5]:
print("TRAIN COLUMNS")
print(train.columns.tolist())

print("\nTEST COLUMNS")
print(test.columns.tolist())

print("\nSTORE COLUMNS")
print(store.columns.tolist())

TRAIN COLUMNS
['Store', 'DayOfWeek', 'Date', 'Sales', 'Customers', 'Open', 'Promo', 'StateHoliday', 'SchoolHoliday']

TEST COLUMNS
['Id', 'Store', 'DayOfWeek', 'Date', 'Open', 'Promo', 'StateHoliday', 'SchoolHoliday']

STORE COLUMNS
['Store', 'StoreType', 'Assortment', 'CompetitionDistance', 'CompetitionOpenSinceMonth', 'CompetitionOpenSinceYear', 'Promo2', 'Promo2SinceWeek', 'Promo2SinceYear', 'PromoInterval']


In [6]:
train.head()

,Store,DayOfWeek,Date,Sales,Customers,Open,Promo,StateHoliday,SchoolHoliday
0,1,5,2015-07-31,5263,555,1,1,0,1
1,2,5,2015-07-31,6064,625,1,1,0,1
2,3,5,2015-07-31,8314,821,1,1,0,1
3,4,5,2015-07-31,13995,1498,1,1,0,1
4,5,5,2015-07-31,4822,559,1,1,0,1


In [7]:
store.head()

,Store,StoreType,Assortment,CompetitionDistance,CompetitionOpenSinceMonth,CompetitionOpenSinceYear,Promo2,Promo2SinceWeek,Promo2SinceYear,PromoInterval
0,1,c,a,1270.0,9.0,2008.0,0,NaN,NaN,NaN
1,2,a,a,570.0,11.0,2007.0,1,13.0,2010.0,"Jan,Apr,Jul,Oct"
2,3,a,a,14130.0,12.0,2006.0,1,14.0,2011.0,"Jan,Apr,Jul,Oct"
3,4,c,c,620.0,9.0,2009.0,0,NaN,NaN,NaN
4,5,a,a,29910.0,4.0,2015.0,0,NaN,NaN,NaN


In [8]:
print("Training period:")
print("Start:", train["Date"].min())
print("End:", train["Date"].max())

print("\nTesting period:")
print("Start:", test["Date"].min())
print("End:", test["Date"].max())

Training period:
Start: 2013-01-01
End: 2015-07-31

Testing period:
Start: 2015-08-01
End: 2015-09-17


In [9]:
print("Number of unique stores:", train["Store"].nunique())
print("Number of unique dates:", train["Date"].nunique())

Number of unique stores: 1115
Number of unique dates: 942


In [10]:
print("Missing values in train:")
print(train.isnull().sum())

print("\nMissing values in test:")
print(test.isnull().sum())

print("\nMissing values in store:")
print(store.isnull().sum())

Missing values in train:
Store            0
DayOfWeek        0
Date             0
Sales            0
Customers        0
Open             0
Promo            0
StateHoliday     0
SchoolHoliday    0
dtype: int64

Missing values in test:
Id                0
Store             0
DayOfWeek         0
Date              0
Open             11
Promo             0
StateHoliday      0
SchoolHoliday     0
dtype: int64

Missing values in store:
Store                          0
StoreType                      0
Assortment                     0
CompetitionDistance            3
CompetitionOpenSinceMonth    354
CompetitionOpenSinceYear     354
Promo2                         0
Promo2SinceWeek              544
Promo2SinceYear              544
PromoInterval                544
dtype: int64


In [11]:
print("Duplicate rows in train:", train.duplicated().sum())
print("Duplicate rows in test:", test.duplicated().sum())
print("Duplicate rows in store:", store.duplicated().sum())

Duplicate rows in train: 0
Duplicate rows in test: 0
Duplicate rows in store: 0


In [12]:
train.info()

<class 'pandas.DataFrame'>
RangeIndex: 1017209 entries, 0 to 1017208
Data columns (total 9 columns):
 #   Column         Non-Null Count    Dtype 
---  ------         --------------    ----- 
 0   Store          1017209 non-null  int64 
 1   DayOfWeek      1017209 non-null  int64 
 2   Date           1017209 non-null  str   
 3   Sales          1017209 non-null  int64 
 4   Customers      1017209 non-null  int64 
 5   Open           1017209 non-null  int64 
 6   Promo          1017209 non-null  int64 
 7   StateHoliday   1017209 non-null  object
 8   SchoolHoliday  1017209 non-null  int64 
dtypes: int64(7), object(1), str(1)
memory usage: 69.8+ MB


In [13]:
train["Sales"].describe()

count    1.017209e+06
mean     5.773819e+03
std      3.849926e+03
min      0.000000e+00
25%      3.727000e+03
50%      5.744000e+03
75%      7.856000e+03
max      4.155100e+04
Name: Sales, dtype: float64

In [14]:
print("Stores with zero sales:", (train["Sales"] == 0).sum())
print("Closed store observations:", (train["Open"] == 0).sum())

Stores with zero sales: 172871
Closed store observations: 172817


In [15]:
print(train["StateHoliday"].value_counts())
print("\nPromo:")
print(train["Promo"].value_counts())

print("\nOpen:")
print(train["Open"].value_counts())

StateHoliday
0    855087
0    131072
a     20260
b      6690
c      4100
Name: count, dtype: int64

Promo:
Promo
0    629129
1    388080
Name: count, dtype: int64

Open:
Open
1    844392
0    172817
Name: count, dtype: int64


In [16]:
zero_sales_open = train[
    (train["Sales"] == 0) & (train["Open"] == 1)
]

print("Open stores with zero sales:", len(zero_sales_open))
zero_sales_open.head()

Open stores with zero sales: 54


,Store,DayOfWeek,Date,Sales,Customers,Open,Promo,StateHoliday,SchoolHoliday
86825,971,5,2015-05-15,0,0,1,0,0,1
142278,674,4,2015-03-26,0,0,1,0,0,0
196938,699,4,2015-02-05,0,0,1,1,0,0
322053,708,3,2014-10-01,0,0,1,1,0,0
330176,357,1,2014-09-22,0,0,1,0,0,0


## Initial Dataset Observations

- The training dataset contains 1,017,209 observations across 1,115 stores.
- Historical training data covers January 2013 to July 2015.
- The test period covers August 2015 to September 2015.
- The target variable is `Sales`, representing daily store sales.
- Average daily sales are approximately 5,774, with a maximum observed value of 41,551.
- There are no duplicate rows in the train, test, or store datasets.
- The training dataset contains no missing values.
- The test dataset contains 11 missing values in the `Open` column.
- The store dataset contains missing values related to Promo2 information.
- Most zero-sales observations correspond to stores that were closed.
- 54 observations have `Open = 1` but both `Sales = 0` and `Customers = 0`; these will be investigated rather than automatically removed.
- Promotions are active in a substantial portion of store-day observations.